src/pcp/atop/photoproc.c

```diff
@@ -34,6 +34,54 @@ extern regex_t  envregex;
 
 #define ABBENVLEN	16
 
+// sort function by TGID
+static int
+thread_sorter(const void *a, const void *b)
+{
+	struct tstat *ta = (struct tstat *)a;
+	struct tstat *tb = (struct tstat *)b;
+	register int64_t a64, b64;
+	a64 = ((int64_t)ta->gen.pid) << 32 | (int64_t)ta->gen.tgid;
+	b64 = ((int64_t)tb->gen.pid) << 32 | (int64_t)tb->gen.tgid;
+	return a64 - b64;
+}
+
+static void
+cpu_accumulation(struct tstat **tasks, int count)
+{
+	qsort(*tasks, count, sizeof(struct tstat), thread_sorter);
+
+	struct tstat *parent = NULL;
+	struct tstat *current;
+	int64_t cpu_use_u, cpu_use_s = 0;
+	int p_index;
+	for (int i=0; i < count; i++)
+	{
+		current = &(*tasks)[i];
+		if (parent == NULL)
+		{
+			parent = current;
+			cpu_use_u = current->cpu.utime;
+			cpu_use_s = current->cpu.stime;
+		}
+		else if (parent->gen.tgid != current->gen.tgid)
+		{
+			(&(*tasks)[p_index])->cpu.utime = cpu_use_u;
+			(&(*tasks)[p_index])->cpu.stime = cpu_use_s;
+			parent = current;
+			cpu_use_u = current->cpu.utime;
+			cpu_use_s = current->cpu.stime;
+		}
+		else
+		{
+			cpu_use_u += current->cpu.utime;
+			cpu_use_s += current->cpu.stime;
+		}
+		if (current->gen.pid == current->gen.tgid)
+			p_index = i;
+	}
```

Comment:
There's a bug here in that the last group doesn't get correctly accounted (i.e. the case where we reach the end of the array - 'i == count' - we don't go through the code path that accumulates the CPU usage.  We'll need an update outside the loop to handle that case.

Reply:
Yes, I noticed that the last set of threads aren't accumulated also.  Probably need something like the following immediately after the loop:

```
  if (parent != NULL)
     parent->cpu.utime = cpu_use_utime;
     parent->cpu.stime = cpu_use_stime;

```